aks-node-controller/app.go

```diff
@@ -58,27 +58,30 @@ func (a *App) run(ctx context.Context, args []string) error {
 	}
 	switch args[1] {
 	case "provision":
+		var err error
+		// Ensure provision-wait is unblocked if we error before provisioning scripts create provision.complete
+		defer a.notifyProvisionFailure(&err)
```

Comment:
I'm not sure if using `defer` here is the best approach. Seems easy to make a mistake by something like `return a.Provision(ctx, ProvisionFlags{ProvisionConfig: *provisionConfig})`

Maybe a more robust solution would be extracting a function returning an error? And handle the error properly?

Reply:
That's a good call out. I've updated the pattern in case people will forget to always assign error to `err`.